simulation.cpp

```diff
@@ -197,8 +203,34 @@ void assign_inputs(population &p, const std::vector<double> &inputs)
 {
   std::vector<individual> vec_inds;
   for(auto& ind : p.get_inds()){
-      ind.set_input(inputs);
+      ind.assign_input(inputs);
+    }
+}
+
+bool all_individuals_have_same_input(const simulation &s)
+{
+  population p = s.get_pop();  //Would it be prefered to refer to a all_individuals_have_same_input population-level function instead?
```

Comment:
Yes you could just make it another function `all_individuals_have_same_input` that takes a population as argument and do all the work there, and then overload that function with a "wrapper" function that takes a sim as an argument, extracts the population and then calls the population level function to check for real.
Actually, now that I think about it, that is how I would like it, if you could do it it would be great!

```suggestion
return all_individuals_have_same_input(s.get_pop()); //and then write in population.cpp the function that does all the work (basically the body of this function)
```